tests/Unit/CrudPanel/CrudPanelFakeFieldsTest.php

```diff
@@ -0,0 +1,149 @@
+    {
+        $this->markTestIncomplete('Not correctly implemented');
+
+        $this->crudPanel->addFields($this->fakeFieldsArray);
+
+        // TODO: the compactFakeFields method fails when a fake field has the same value for the 'name' and 'store_in'
+        //       attributes (in our case, the tags field in the fake field array)
+        $compactedFakeFields = $this->crudPanel->compactFakeFields($this->fakeFieldsInputData, 'create');
+
+        $this->assertEquals($this->expectedInputDataWithCompactedFakeFields, $compactedFakeFields);
+    }
+
+    public function testCompactFakeFieldsFromUpdateForm()
+    {
+        $this->markTestIncomplete('Not correctly implemented');
+
+        $article = DB::table('articles')->where('id', 1)->first();
+        $this->crudPanel->setModel(Article::class);
+        $this->crudPanel->addFields($this->fakeFieldsArray, 'update');
+
+        // TODO: the compactFakeFields method fails when a fake field has the same value for the 'name' and 'store_in'
+        //       attributes (in our case, the tags field in the fake field array)
+        $compactedFakeFields = $this->crudPanel->compactFakeFields($this->fakeFieldsInputData, 'update', $article->id);
+
+        $this->assertEquals($this->expectedInputDataWithCompactedFakeFields, $compactedFakeFields);
+    }
+
+    public function testCompactFakeFieldsFromUpdateFormWithoutId()
+    {
+        $this->setExpectedException(ModelNotFoundException::class);
+
+        $this->crudPanel->setModel(Article::class);
+
+        $this->crudPanel->addFields($this->fakeFieldsArray, 'update');
+        $compactedFakeFields = $this->crudPanel->compactFakeFields($this->fakeFieldsInputData, 'update');
+
+        $this->assertEquals($this->expectedInputDataWithCompactedFakeFields, $compactedFakeFields);
+    }
+
+    public function testCompactFakeFieldsFromUpdateFormWithUnknownId()
+    {
+        $this->setExpectedException(ModelNotFoundException::class);
+
+        $unknownId = DB::getPdo()->lastInsertId() + 1;
+        $this->crudPanel->setModel(Article::class);
+        $this->crudPanel->addFields($this->fakeFieldsArray, 'update');
+
+        $compactedFakeFields = $this->crudPanel->compactFakeFields($this->fakeFieldsInputData, 'update', $unknownId);
+
+        $this->assertEquals($this->expectedInputDataWithCompactedFakeFields, $compactedFakeFields);
+    }
+
+    public function testCompactFakeFieldsFromEmptyRequest()
+    {
+        $compactedFakeFields = $this->crudPanel->compactFakeFields([]);
+
+        $this->assertEmpty($compactedFakeFields);
+    }
+
+    public function testCompactFakeFieldsFromRequestWithNoFakes()
+    {
+        $compactedFakeFields = $this->crudPanel->compactFakeFields($this->noFakeFieldsInputData);
+
+        $this->assertEquals($this->noFakeFieldsInputData, $compactedFakeFields);
+    }
+
+    public function testCompactFakeFieldsFromUnknownForm()
+    {
+        $this->markTestIncomplete('Not correctly implemented');
```

Comment:
If the method *should* throw an exceptoin, but it doesn't, and the test correctly tests that, how is it that the test is incomplete?

(i.e. the test is complete and the thing it's testing has a bug - unless I misunderstand the below comment).

Reply:
Haha - guess the naming is a bit off. The test is correctly implemented, but the feature isn't, I guess :-)